core/schemas/provider.go

```diff
@@ -123,6 +123,114 @@ func (ar *AllowedRequests) IsOperationAllowed(operation Operation) bool {
 	}
 }
 
+// GetProviderCapabilities returns the capabilities for a given provider
+func GetProviderCapabilities(provider ModelProvider) AllowedRequests {
```

Comment:
@TejasGhatte pls change its name to get GetBaseProviderCapabilities cause it only contains the base providers, not all